Move the Teleport IdP section to Identity Governance

ptgott opened the pull request:
Closes #55102

- Move the `idps` directory from `docs/pages/admin-guides/access-controls` to `docs/pages/enroll-resources/application-access`.
- Update internal links.
- Add redirects.

ptgott committed:
Move the Teleport IdP section to Enroll Resources

Closes #55102

- Move the `idps` directory from
  `docs/pages/admin-guides/access-controls` to
  `docs/pages/enroll-resources/application-access`.
- Update internal links.
- Add redirects.

zmb3 commented:
Hmm, are we sure this is the right thing to do? `Enroll Resources` is mostly Teleport Zero Trust, and Teleport as an IdP is part of Identity Governance, right @garofalof?

flyinghermit commented:
My 2c: any user who is not aware or is ignorant of Teleport product terminology (which also changes frequently) will be inclined to look for "adding x to Teleport" docs under "Enroll Resources".

Given that assumption, the two paths I can recommend:
1. Add "Teleport SAML IdP" under new IGS section but list all "add x saml app to Teleport" guides 
under "Enroll Resources > Applications > SAML application". The IGS page can link enrollment guides to "Enroll Resources"

2. Add a landing page under "Enroll Resources > Applications > SAML application", that links to IGS section. So users learn that they can add SAML apps too.

garofalof commented:
> Hmm, are we sure this is the right thing to do? `Enroll Resources` is mostly Teleport Zero Trust, and Teleport as an IdP is part of Identity Governance, right @garofalof?

yes, Teleport as an IdP is Identity Governance

flyinghermit reviewed (approved):
Thank you for this change, Paul.

ptgott commented:
@zmb3 @garofalof @flyinghermit Another possibility would be to move the `idps` directory into `identity-governance` and, in the index page for enrolling applications (`docs/pages/enroll-resources/application-access/application-access.mdx`), indicate that you can enroll SAML apps by configuring Teleport to be an IdP using Identity Governance. 

This way, we can continue to place all Identity Governance docs under one URL path while letting visitors to the "Enroll Applications" section know that they can use the IdP approach. How does that sound?

flyinghermit commented:
Sounds good to me, as long as there is an easy to find reference under enroll resources page that points to SAML docs under IGS

ptgott committed:
Move the idps section into Identity Governance

The Teleport SAML IdP is an Identity Governance feature. Make sure that
all Identity Governance features fall within the appropriate section of
the docs.

Add a paragraph to the "Enroll Applications" index page indicating that
you can enroll SAML applications using the Teleport IdP.
